src/components/Badge/Badge.js

```diff
@@ -2,35 +2,47 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 import { THEMES, getThemeClass } from '../../utils';
+import { applyDecorators, decoratorProp } from '../../utils';
 
 const Badge = (props) => {
   const {
+    children,
     className,
-    label,
+    flavor,
     theme,
     ...rest
   } = props;
 
   const sldsClasses = [
     'slds-badge',
     ...getThemeClass(theme),
+    applyDecorators(flavor, 'badge'),
     className,
   ];
 
-  return (<span {...rest} className={cx(sldsClasses)}>{label}</span>);
+  return <span {...rest} className={cx(sldsClasses)}>{children}</span>;
 };
 
 Badge.defaultProps = {
   className: null,
+  flavor: null,
   theme: null,
 };
 
 Badge.propTypes = {
+  /** Either a string or a string & an icon */
+  children: PropTypes.node.isRequired,
   /** (optional) className */
   className: PropTypes.string,
-  /** Text that will be displayed */
-  label: PropTypes.string.isRequired,
-  /** SLDS Theme */
+  /**
+   * Badge flavor: array of flavors, you can also provide a single flavor string.
+   * Flavors: inverse, lightest
+  */
+  flavor: decoratorProp([
+    'inverse',
+    'lightest'
+  ]),
+  /** DEPRECTATED: SLDS Theme */
```

Comment:
With the breaking changes already introduced, what about removing the theme immediately instead of deprecating?

Reply:
We still need it for colored badges that we are currently using, not possible with new badges